examples/dreambooth/train_dreambooth_lora.py

```diff
@@ -910,27 +910,23 @@ def save_model_hook(models, weights, output_dir):
         )
 
     def load_model_hook(models, input_dir):
-        # Note we DON'T pass the unet and text encoder here an purpose
-        # so that the we don't accidentally override the LoRA layers of
-        # unet_lora_layers and text_encoder_lora_layers which are stored in `models`
-        # with new torch.nn.Modules / weights. We simply use the pipeline class as
-        # an easy way to load the lora checkpoints
-        temp_pipeline = DiffusionPipeline.from_pretrained(
-            args.pretrained_model_name_or_path,
-            revision=args.revision,
-            torch_dtype=weight_dtype,
-        )
-        temp_pipeline.load_lora_weights(input_dir)
-
-        # load lora weights into models
-        models[0].load_state_dict(AttnProcsLayers(temp_pipeline.unet.attn_processors).state_dict())
-        if len(models) > 1:
-            models[1].load_state_dict(AttnProcsLayers(temp_pipeline.text_encoder_lora_attn_procs).state_dict())
+        lora_weights = torch.load(os.path.join(input_dir, LORA_WEIGHT_NAME))
+        unet_weights = {}
+        text_encoder_weights = {}
+
+        for k, v in lora_weights.items():
+            model = k.split(".")[0]
+            k = k.split(".")[1:]
+
+            if model == UNET_NAME:
+                unet_weights[k] = v
+            elif model == TEXT_ENCODER_NAME:
+                text_encoder_weights[k] = v
+            else:
+                raise ValueError(f"unknown model name {model}")
 
-        # delete temporary pipeline and pop models
-        del temp_pipeline
-        for _ in range(len(models)):
-            models.pop()
+        unet.load_state_dict(unet_weights, strict=False)
+        text_encoder.load_state_dict(text_encoder_weights, strict=False)
```

Comment:
Similar to the modifying the text encoder case, we should be able to just load the weights from a serialized state dict without instantiating a pipeline